### ISSUE TITLE
[Backtrace] Crash when hitting `g` to reply-all in a dynamic folder resulting from a notmuch search

### ISSUE THREAD

**Issue by RichiH** [neomutt/neomutt#207]
Affects mutt 1.7.0-6 on Debian testing.

As per topic, what I did:

* Install mutt-dbgsym on Debian testing
* `gdb mutt`; run
* Use mutt for some time
* Search for mail
* Open mail
* Go to attachment view
* Open attached PDF (Microsoft legacy `=?iso-8859...` encoding for the attachment name) with okular
* Keep PDF open for ~1 hour
* Close Okular normally
* hit `g` in email view

This leads to:

````
(gdb) bt
#0  0x00005555555d38f6 in mutt_fetch_recips (out=out@entry=0x55555879cb90, in=in@entry=0x0, flags=flags@entry=3) at ../../send.c:586
#1  0x00005555555d5b53 in envelope_defaults (flags=3, cur=0x5555579f6b00, ctx=0x5555579154a0, env=0x55555879cb90) at ../../send.c:800
#2  ci_send_message (flags=<optimized out>, flags@entry=3, msg=<optimized out>, msg@entry=0x0, tempfile=tempfile@entry=0x0, ctx=0x5555579154a0, cur=<optimized out>) at ../../send.c:1456
#3  0x00005555555bd227 in mutt_pager (banner=banner@entry=0x0, fname=fname@entry=0x7fffffffc4d0 "/tmp/mutt-Mac-rih-1000-10017-5729660799768819081", flags=<optimized out>, flags@entry=66, extra=extra@entry=0x7fffffffc4a0) at ../../pager.c:2812
#4  0x000055555557c165 in mutt_display_message (cur=0x5555579f6b00) at ../../commands.c:225
#5  0x000055555558c79b in mutt_index_menu () at ../../curs_main.c:1908
#6  0x000055555556c7fa in main (argc=1, argv=<optimized out>) at ../../main.c:876
(gdb) 
````

**Comment by flatcap** [comment]
I've just committed some code that may fix this problem: a75b414
The only time you can get an invalid header pointer is if the email you're looking at is deleted/moved.
This, I managed to reproduce and it caused the same sorts of "random" crashes we've been seeing.
If you've the time, please give this a test.  If not, I'm planning a release for Friday.
Thanks.

**Comment by RichiH** [comment]
I am on the road a lot this week, so I might need to wait for the release.
I never delete or move any mail, but as it's usually in search results,
those mails might seem to have been deleted.

Thanks! I will run the new version in gdb for the foreseeable future as
well.

Richard

Sent by mobile; excuse my brevity.
